modules/reactor/include/meshgenerators/PeripheralRingMeshGenerator.h

```diff
@@ -82,4 +96,28 @@ class PeripheralRingMeshGenerator : public PolygonMeshGeneratorBase
    * @return whether the boundary is the external boundary of the given mesh
    */
   bool isExternalBoundary(ReplicatedMesh & mesh, const boundary_id_type bid) const;
+
+  /**
+   * Define node positions of the inner boundary layer that is conformal to the input mesh's
+   * external boundary.
+   * @param input_ext_node_num number of nodes on the external boundary of the input mesh
+   * @param input_bdry_angles list of angles (in rad) formed by three neighboring nodes on the
+   * external boundary of the input mesh
+   * @param ref_inner_bdry_surf Reference outmost layer (surface) points of the inner boundary layer

```

Comment:
so from  ref_inner_bdry_surf I can get input_bdry_angles and if I know origin_pt I can also get ref_inner_bdry_azi azimuthal ?

Reply:
Yes, I may eliminate one